src/test/java/org/springframework/samples/petclinic/owner/OwnerControllerTests.java

```diff
@@ -174,5 +181,36 @@ public void testShowOwner() throws Exception {
             .andExpect(model().attribute("owner", hasProperty("telephone", is("6085551023"))))
             .andExpect(view().name("owners/ownerDetails"));
     }
+    
+    @Test
+    public void testMockOwnerRepository() {
+    	//Use the fake repository to break dependency of owner controller on owners repository
+    	FakeOwnerRepository fakeOwners = new FakeOwnerRepository();
+
+    	//Creation of owner controller with fake owner repository and instead of mocked owners by OwnerMVC
+    	OwnerController ownerController = new OwnerController(fakeOwners);
+    	
+    	//Mocking mapping-related objects
+    	BindingResult result = mock(BindingResult.class);
+    	Model model = mock(Model.class);
+    	
+    	//internally this will use the fake owner's repository's save method
+    	ownerController.processCreationForm(george, result);
+    	assertEquals(george.getId().intValue(), fakeOwners.getOwnerId(george));
+      	 
+    	//internally this will use the fake owner's repository's save method
+    	ownerController.processUpdateOwnerForm(george, result, george.getId());
+    	assertEquals(george.getId().intValue(), fakeOwners.getOwnerId(george));
```

Comment:
Perhaps use `findById` or `findByLastName` for the assertion

Reply:
Will make this change.